Error with invalid discovery messages

This error is reported during discovery:
```
Jun 13 06:55:26 ip-172-31-23-167 crawler[3231927]: 06:55:26.613 [DefaultDispatcher-worker-1] ERROR org.apache.tuweni.devp2p.DiscoveryService - ÐΞVp2p discovery 957395249: unexpected error during packet handling
Jun 13 06:55:26 ip-172-31-23-167 crawler[3231927]: com.google.common.util.concurrent.UncheckedExecutionException: java.lang.IllegalArgumentException: Invalid point coordinates
Jun 13 06:55:26 ip-172-31-23-167 crawler[3231927]: #011at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2050)
Jun 13 06:55:26 ip-172-31-23-167 crawler[3231927]: #011at com.google.common.cache.LocalCache.get(LocalCache.java:3952)
Jun 13 06:55:26 ip-172-31-23-167 crawler[3231927]: #011at com.google.common.cache.LocalCache$LocalManualCache.get(LocalCache.java:4871)
Jun 13 06:55:26 ip-172-31-23-167 crawler[3231927]: #011at org.apache.tuweni.devp2p.DevP2PPeerRoutingTable.hashForId(PeerRoutingTable.kt:100)
Jun 13 06:55:26 ip-172-31-23-167 crawler[3231927]: #011at org.apache.tuweni.devp2p.DevP2PPeerRoutingTable.nearest(PeerRoutingTable.kt:92)
Jun 13 06:55:26 ip-172-31-23-167 crawler[3231927]: #011at org.apache.tuweni.devp2p.CoroutineDiscoveryService.neighbors(DiscoveryService.kt:863)
Jun 13 06:55:26 ip-172-31-23-167 crawler[3231927]: #011at org.apache.tuweni.devp2p.CoroutineDiscoveryService.handleFindNode(DiscoveryService.kt:610)
Jun 13 06:55:26 ip-172-31-23-167 crawler[3231927]: #011at org.apache.tuweni.devp2p.CoroutineDiscoveryService$handleFindNode$1.invokeSuspend(DiscoveryService.kt)
Jun 13 06:55:26 ip-172-31-23-167 crawler[3231927]: #011at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
Jun 13 06:55:26 ip-172-31-23-167 crawler[3231927]: #011at kotlinx.coroutines.internal.ScopeCoroutine.afterResume(Scopes.kt:32)
Jun 13 06:55:26 ip-172-31-23-167 crawler[3231927]: #011at kotlinx.coroutines.AbstractCoroutine.resumeWith(AbstractCoroutine.kt:113)
Jun 13 06:55:26 ip-172-31-23-167 crawler[3231927]: #011at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)
Jun 13 06:55:26 ip-172-31-23-167 crawler[3231927]: #011at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
Jun 13 06:55:26 ip-172-31-23-167 crawler[3231927]: #011at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571)
Jun 13 06:55:26 ip-172-31-23-167 crawler[3231927]: #011at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
Jun 13 06:55:26 ip-172-31-23-167 crawler[3231927]: #011at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678)
Jun 13 06:55:26 ip-172-31-23-167 crawler[3231927]: #011at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665)
Jun 13 06:55:26 ip-172-31-23-167 crawler[3231927]: Caused by: java.lang.IllegalArgumentException: Invalid point coordinates
Jun 13 06:55:26 ip-172-31-23-167 crawler[3231927]: #011at org.bouncycastle.math.ec.ECCurve.validatePoint(ECCurve.java:125)
Jun 13 06:55:26 ip-172-31-23-167 crawler[3231927]: #011at org.bouncycastle.math.ec.ECCurve.decodePoint(ECCurve.java:418)
Jun 13 06:55:26 ip-172-31-23-167 crawler[3231927]: #011at org.apache.tuweni.crypto.SECP256K1$PublicKey.asEcPoint(SECP256K1.java:707)
Jun 13 06:55:26 ip-172-31-23-167 crawler[3231927]: #011at org.apache.tuweni.devp2p.EthereumNodeRecord$Companion.nodeId(EthereumNodeRecord.kt:58)
Jun 13 06:55:26 ip-172-31-23-167 crawler[3231927]: #011at org.apache.tuweni.devp2p.DevP2PPeerRoutingTable$hashForId$1.call(PeerRoutingTable.kt:100)
Jun 13 06:55:26 ip-172-31-23-167 crawler[3231927]: #011at org.apache.tuweni.devp2p.DevP2PPeerRoutingTable$hashForId$1.call(PeerRoutingTable.kt:69)
Jun 13 06:55:26 ip-172-31-23-167 crawler[3231927]: #011at com.google.common.cache.LocalCache$LocalManualCache$1.load(LocalCache.java:4876)
Jun 13 06:55:26 ip-172-31-23-167 crawler[3231927]: #011at com.google.common.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3528)
Jun 13 06:55:26 ip-172-31-23-167 crawler[3231927]: #011at com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2277)
Jun 13 06:55:26 ip-172-31-23-167 crawler[3231927]: #011at com.google.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2154)
Jun 13 06:55:26 ip-172-31-23-167 crawler[3231927]: #011at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2044)
Jun 13 06:55:26 ip-172-31-23-167 crawler[3231927]: #011... 16 common frames omitted
```

--- a/devp2p/src/main/kotlin/org/apache/tuweni/devp2p/PeerRoutingTable.kt
+++ b/devp2p/src/main/kotlin/org/apache/tuweni/devp2p/PeerRoutingTable.kt
@@ -18,8 +18,10 @@ package org.apache.tuweni.devp2p
 
 import com.google.common.cache.Cache
 import com.google.common.cache.CacheBuilder
+import com.google.common.util.concurrent.UncheckedExecutionException
 import org.apache.tuweni.crypto.SECP256K1
 import org.apache.tuweni.kademlia.KademliaRoutingTable
+import java.lang.IllegalArgumentException
 
 /**
  * A routing table for ÐΞVp2p peers.
@@ -98,8 +100,11 @@ internal class DevP2PPeerRoutingTable(selfId: SECP256K1.PublicKey) : PeerRouting
   private fun hashForId(id: SECP256K1.PublicKey): ByteArray? {
     try {
       return idHashCache.get(id) { EthereumNodeRecord.nodeId(id).toArrayUnsafe() }
-    } catch (e: IllegalArgumentException) {
-      return null
+    } catch (e: UncheckedExecutionException) {
+      if (e.cause is IllegalArgumentException) {
+        return null
+      }
+      throw e
     }
   }
 }

--- a/devp2p/src/test/kotlin/org/apache/tuweni/devp2p/v5/PeerRoutingTableTest.kt
+++ b/devp2p/src/test/kotlin/org/apache/tuweni/devp2p/v5/PeerRoutingTableTest.kt
@@ -1,0 +1,37 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.tuweni.devp2p.v5
+
+import org.apache.tuweni.bytes.Bytes
+import org.apache.tuweni.crypto.SECP256K1
+import org.apache.tuweni.devp2p.DevP2PPeerRoutingTable
+import org.apache.tuweni.junit.BouncyCastleExtension
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+
+@ExtendWith(BouncyCastleExtension::class)
+class PeerRoutingTableTest {
+
+  @Test
+  fun invalidPublicKey() {
+    val routingTable = DevP2PPeerRoutingTable(SECP256K1.KeyPair.random().publicKey())
+    val invalidPublicKey = SECP256K1.PublicKey.fromBytes(Bytes.repeat(0, 64))
+    val peers = routingTable.nearest(invalidPublicKey, 3)
+    assertEquals(0, peers.size)
+  }
+}

